sdk/storage/azdatalake/common.go

```diff
@@ -0,0 +1,81 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+package azdatalake
+
+import (
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/blob"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/container"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azdatalake/internal/generated"
+)
+
+const SnapshotTimeFormat = "2006-01-02T15:04:05.0000000Z07:00"
+
+// FilesystemAccessConditions identifies container-specific access conditions which you optionally set.
+type FilesystemAccessConditions struct {
+	ModifiedAccessConditions *ModifiedAccessConditions
+	LeaseAccessConditions    *LeaseAccessConditions
+}
+
+// FormatContainerAccessConditions formats FilesystemAccessConditions into container's LeaseAccessConditions and ModifiedAccessConditions.
+func FormatContainerAccessConditions(b *FilesystemAccessConditions) *container.AccessConditions {
```

Comment:
Can these be moved to an internal package so they don't need to be exported?  I assume these kinds of functions aren't intended to be directly called by customers.

Reply:
Makes sense, moved.